File: src/Visitor/Visitor.java
package Visitor;

import Composite.Directory;
import Composite.AComponent;
import Factory.DataFrame;


public abstract class Visitor {

    public abstract void visit(DataFrame file);

    /**
     * Method that is visiting the files
     *
     * @param dir directory
     */
    public void visit(Directory dir) {
        for(AComponent elem: dir.getChildren() )
            elem.accept(this);
    }

}

File: src/Scala/Visitor/Visitor.scala
package Scala.Visitor

import Factory.DataFrame
import Scala.Composite.DirectoryScala

abstract class Visitor{

  def visit(file: DataFrame): Unit

  /**
   * Method that is visiting the files
   *
   * @param dir directory
   */
  def visit(dir: DirectoryScala): Unit = {
    for (elem <- dir.children) {
      elem.accept(this)

    }
  }


}
